client/app/src/main/java/com/example/mrides/Activity/CreateRouteActivity.java

```diff
@@ -376,4 +386,75 @@ public void responseReceived(String response) {
         successObtainDirection(route);
     }
 
+    public void matchRoute(List<LatLng> routeOfUser) {
+
+        for (User user : userOnMapCatalog) {
+            ArrayList<Route> passengerRoutes = user.getRoutes();
+
+            for (Route route : passengerRoutes) {
+                LatLng pickUp = route.getStartLocation();
+                LatLng drop = route.getEndLocation();
+                int passengerRouteId = route.getId();
+                boolean pickUpBool = false;
+                boolean goToEnd = false;
+                int i = 0;
+
+                while (i < routeOfUser.size() && pickUpBool == false) {
+                    LatLng pointInPoly = routeOfUser.get(i);
+//                    System.out.println("Distance " + (distance(pointInPoly.latitude, pointInPoly.longitude
+//                    , drop.latitude, drop.longitude) <= 0.1) + "Route " + passengerRouteId);
+                    if (validateDistance(pickUp, pointInPoly) && goToEnd == false) {
+                        goToEnd = true;
+                        i++;
+                    }
+
+                    if (validateDistance(drop, pointInPoly) && goToEnd == true) {
+                        for ( int key : matchedMarkers.keySet()) {
+                            if(key == passengerRouteId) {
+                                Marker marker = matchedMarkers.get(key);
+                                marker.setIcon(BitmapDescriptorFactory.defaultMarker(BitmapDescriptorFactory.HUE_ROSE));
+                                pickUpBool = true;
+                                break;
+                            }
+                        }
+                    }
+                    i++;
+                }
+
+            }
+
+        }
+    }
+
+    /** calculates the distance between two locations in MILES */
+    private double distance(double lat1, double lng1, double lat2, double lng2) {
+
+        double earthRadius = 3958.75; // in miles, change to 6371 for kilometer output
+
+        double dLat = Math.toRadians(lat2-lat1);
+        double dLng = Math.toRadians(lng2-lng1);
+
+        double sindLat = Math.sin(dLat / 2);
+        double sindLng = Math.sin(dLng / 2);
+
+        double a = Math.pow(sindLat, 2) + Math.pow(sindLng, 2)
+                * Math.cos(Math.toRadians(lat1)) * Math.cos(Math.toRadians(lat2));
+
+        double c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1-a));
+
+        double dist = earthRadius * c;
+
+        return dist; // output distance, in MILES
+    }
+
+    public boolean validateDistance(LatLng passengerLocation, LatLng userLocation) {
```

Comment:
Add comment to explain what does "0.1" signify in the matching algorithm. For example: distance smaller than 160 meters.